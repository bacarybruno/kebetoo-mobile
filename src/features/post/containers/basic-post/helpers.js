export const POST_TYPES = {
  AUDIO: 'audio',
  IMAGE: 'image',
  TEXT: 'text',
  REPOST: 'repost',
  VIDEO: 'video',
};

export const getPostType = (post) => {
  if (post.repost) {
    return POST_TYPES.REPOST;
  }
  if (post.audio && post.audio.url) {
    return POST_TYPES.AUDIO;
  }
  if (post.image && post.image.url) {
    return POST_TYPES.IMAGE;
  }
  if (post.video && post.video.url) {
    return POST_TYPES.VIDEO;
  }
  if (post.content && post.content.length > 0) {
    return POST_TYPES.TEXT;
  }
  return null;
};
