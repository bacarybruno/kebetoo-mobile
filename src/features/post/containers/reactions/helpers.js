export const REACTION_TYPES = {
  LIKE: 'like',
  DISLIKE: 'dislike',
  COMMENT: 'comment',
  LOVE: 'love',
  SHARE: 'share',
};
