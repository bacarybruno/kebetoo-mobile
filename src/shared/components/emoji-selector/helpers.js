import { Platform } from 'react-native'

import categories, { fullCategoryNames as categoryNames } from './categories'

const charFromUtf16 = (utf16) => String.fromCodePoint(...utf16.split('-').map((u) => `0x${u}`))
const charFromEmojiObject = (obj) => charFromUtf16(obj.unified)

// filters
const filterSupported = (emoji) => (
  !emoji.obsoleted_by
  && emoji[`has_img_${Platform.OS === 'ios' ? 'apple' : 'google'}`]
  && emoji.category !== 'Skin Tones'
)

// sort
const sortByCategory = (a, b) => (
  categoryNames.indexOf(a.category) - categoryNames.indexOf(b.category)
)
const sortBySortOrder = (a, b) => a.sort_order - b.sort_order
const sortByCategoryAndOrder = (a, b) => {
  const categorySort = sortByCategory(a, b)
  const orderSort = sortBySortOrder(a, b)
  return categorySort || orderSort
}

// map
const mapEmojis = (emoji) => ({
  symbol: charFromEmojiObject(emoji),
  shortNames: emoji.short_names,
  category: emoji.category === 'People & Body' ? 'Smileys & Emotion' : emoji.category,
})

// group
const groupByCategory = (emojis) => {
  const groups = {}
  emojis.forEach((emoji) => {
    if (groups[emoji.category] === undefined) groups[emoji.category] = []
    groups[emoji.category].push(emoji)
  })
  if (groups[categories.history.name] === undefined) groups[categories.history.name] = []
  return groups
}

// history params
const historyStorageKey = '@app/storage/emoji-history'
const historyMaxItems = 40

export default {
  sortByCategoryAndOrder,
  charFromEmojiObject,
  filterSupported,
  groupByCategory,
  mapEmojis,
  history: {
    storageKey: historyStorageKey,
    maxItems: historyMaxItems,
  },
}
