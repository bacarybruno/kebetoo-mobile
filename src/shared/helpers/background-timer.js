import { DeviceEventEmitter, NativeAppEventEmitter, Platform } from 'react-native';
import RNBackgroundTimer from 'react-native-background-timer';

const EventEmitter = Platform.select({
  ios: NativeAppEventEmitter,
  android: DeviceEventEmitter,
});

class BackgroundTimer {
  static setInterval(callback, delay) {
    RNBackgroundTimer.start();
    this.backgroundListener = EventEmitter.addListener('backgroundTimer', () => {
      this.backgroundTimer = RNBackgroundTimer.setInterval(callback, delay);
    });
    return this.backgroundListener;
  }

  static clearInterval(timer) {
    if (timer) timer.remove();
    if (this.backgroundTimer) RNBackgroundTimer.clearInterval(this.backgroundTimer);
    RNBackgroundTimer.stop();
  }
}

export default BackgroundTimer;
