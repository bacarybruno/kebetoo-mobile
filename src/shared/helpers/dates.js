import dayjs from 'dayjs'

export const readableSeconds = (value) => {
  const date = dayjs().startOf('day').second(value)
  return date.format('mm:ss')
}