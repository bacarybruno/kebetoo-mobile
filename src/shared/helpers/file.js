export const mimeTypes = Object.freeze({
  // images
  png: 'image/png',
  jpg: 'image/jpeg',
  jpeg: 'image/jpeg',
  gif: 'image/gif',
  // audio
  mp3: 'audio/mp3',
  m4a: 'audio/mp4',
  aac: 'audio/x-aac',
  opus: 'audio/ogg',
  ogg: 'audio/ogg',
  oga: 'audio/ogg',
  // video
  mov: 'video/quicktime',
  mpg: 'video/mpeg',
  mpeg: 'video/mpeg',
  mp4: 'video/mp4',
  wmv: 'video/x-ms-wmv',
  avi: 'video/x-msvideo',
  webm: 'video/webm',
})

const videoExtensions = Object
  .keys(mimeTypes)
  .filter((mime) => mimeTypes[mime].startsWith('video'))

export const getFileName = (path) => {
  if (typeof path !== 'string') return undefined
  const parts = path.split('/')
  return parts[parts.length - 1]
}

export const getExtension = (filename) => {
  if (typeof filename !== 'string') return undefined
  const parts = filename.split('.')
  return parts[parts.length - 1]
}

export const getMimeType = (filename) => {
  const extension = getExtension(filename)
  return mimeTypes[extension]
}

export const getMediaType = (filename) => {
  const mimeType = getMimeType(filename)
  if (mimeType === undefined) return undefined
  return mimeType.split('/')[0]
}

export const isVideo = (filename) => {
  const extension = getExtension(filename)
  return videoExtensions.includes(extension)
}
