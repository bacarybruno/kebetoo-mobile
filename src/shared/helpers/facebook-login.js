import { LoginManager, AccessToken } from 'react-native-fbsdk'
import auth from '@react-native-firebase/auth'

import { createUser } from '@app/shared/helpers/users'

export const getFacebookPicture = (profileId) => `https://graph.facebook.com/${profileId}/picture?type=large`

// TODO: handle result errors
const facebookLogin = async () => {
  const result = { error: null, data: null }
  try {
    LoginManager.logOut()
    const loginResult = await LoginManager.logInWithPermissions(['public_profile', 'email'])
    const { accessToken } = await AccessToken.getCurrentAccessToken()
    const facebookCredential = auth.FacebookAuthProvider.credential(accessToken)
    const data = await auth().signInWithCredential(facebookCredential)

    const { additionalUserInfo, user } = data
    const { profile } = additionalUserInfo
    const { id: profileId } = profile

    const photoURL = getFacebookPicture(profileId)
    await createUser({ id: user.uid, displayName: user.displayName, photoURL })

    user.updateProfile({ photoURL })

    result.data = data
  } catch (error) {
    result.error = error
  }
  return Promise.resolve(result)
}

export default facebookLogin
