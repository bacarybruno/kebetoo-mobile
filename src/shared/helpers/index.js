import * as dates from './dates';
import * as file from './file';

export { default as generateColor } from './color-generator';
export { dates, file };
