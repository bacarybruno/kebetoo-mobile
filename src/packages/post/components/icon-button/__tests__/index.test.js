import { Animated } from 'react-native'

import setupTest from 'Kebetoo/src/config/jest-setup'
import { readableSeconds } from 'Kebetoo/src/shared/helpers/dates'

import IconButton from '../index'

const givenIconButton = setupTest(IconButton)({
  onPress: jest.fn(),
  name: 'microphone',
  style: {
    elevation: 5,
  },
})

it('renders IconButton', () => {
  const { wrapper } = givenIconButton()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

// TODO: find a better way to test this without having knowledge of implementation details
// ie without relying on jest.spyOn(Animated, 'spring')
describe('animation', () => {
  beforeEach(jest.clearAllMocks)
  it('animates IconButton if active', () => {
    const animateSpring = jest.spyOn(Animated, 'spring')
    givenIconButton({
      activable: true,
      isActive: true,
      text: readableSeconds(100),
    })
    expect(animateSpring).toBeCalledTimes(1)
  })
  it('doesnt animates IconButton if not active', () => {
    const animateSpring = jest.spyOn(Animated, 'spring')
    givenIconButton({
      activable: true,
      isActive: false,
      text: readableSeconds(100),
    })
    expect(animateSpring).not.toBeCalled()
  })
})
