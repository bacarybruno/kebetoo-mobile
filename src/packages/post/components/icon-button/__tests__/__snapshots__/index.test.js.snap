// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IconButton 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 20,
        "left": 20,
        "right": 20,
        "top": 20,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "borderRadius": 20,
        "elevation": 5,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 40,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Object {
          "color": "#0B1B77",
          "fontFamily": "icomoon",
          "fontSize": 18.018018018018015,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
      text="kebeticon-microphone"
    />
  </View>
</View>
`;
